src/js/components/help/downloads.js

```diff
@@ -0,0 +1,323 @@
+        title,
+        isUserOs: osMap[os] === supportedOs
+      });
+      return accu;
+    }, []);
+  }
+  return [{ location: `https://downloads.${location}/${id}/${getVersion(versionInfo, id)}/linux/${id}`, title }];
+};
+
+const multiArchLocationFormatter = ({ tool, versionInfo }) => {
+  const location = getToolLocation(tool);
+  const { archList, id, osList, packageId } = tool;
+  return osList.reduce((accu, os) => {
+    const osArchitectureLocations = archList.map(arch => ({
+      location: `https://downloads.${location}/repos/debian/pool/main/m/${id}/${encodeURIComponent(
+        `${packageId || id}_${getVersion(versionInfo, id)}-1+${os}_${arch}.deb`
+      )}`,
+      title: `${os} - ${arch}`
+    }));
+    accu.push(...osArchitectureLocations);
+    return accu;
+  }, []);
+};
+
+const nonOsLocationFormatter = ({ tool, versionInfo }) => {
+  const location = getToolLocation(tool);
+  const { id, title } = tool;
+  return [
+    {
+      location: `https://downloads.${location}/${id}/${getVersion(versionInfo, id)}/${id}-${getVersion(versionInfo, id)}.tar.xz`,
+      title
+    }
+  ];
+};
+
+const defaultCurlDownload = ({ location, tokens }) => {
+  let headers = `-H "Cookie: JWT=${getToken()}"`;
+  if (tokens.length) {
+    headers = `-H "Authorization: Bearer ${tokens[0]}"`;
+  }
+  return `curl ${headers} -LO ${location}`;
+};
+
+const defaultGitlabJob = ({ location, tokens }) => {
+  const filename = location.substring(location.lastIndexOf('/') + 1);
+  return `
+download:mender-tools:
+  image: curlimages/curl
+  stage: download
+  variables:
+    ${tokens.length ? `MENDER_TOKEN: ${tokens}` : `MENDER_JWT: ${getToken()}`}
+  script:
+    - if [ -n "$MENDER_TOKEN" ]; then
+    - curl -H "Authorization: Bearer $MENDER_TOKEN" -LO ${location}
+    - else
+    - ${defaultCurlDownload({ location, tokens })}
+    - fi
+  artifacts:
+    expire_in: 1w
+    paths:
+      - ${filename}
+`;
+};
+
+const tools = [
+  {
+    id: 'mender',
+    packageId: 'mender-client',
+    title: 'Mender Client Debian package',
+    getLocations: multiArchLocationFormatter,
+    canAccess: () => true,
+    osList: defaultOSVersions,
+    archList: defaultArchitectures
+  },
+  {
+    id: 'mender-artifact',
+    title: 'Mender Artifact',
+    getLocations: defaultLocationFormatter,
+    canAccess: () => true,
+    osList: ['darwin', 'linux']
+  },
+  {
+    id: 'mender-binary-delta',
+    title: 'Mender Binary Delta generator and Update Module',
+    getLocations: nonOsLocationFormatter,
+    isPublic: false,
+    canAccess: ({ isEnterprise, tenantCapabilities }) => isEnterprise || tenantCapabilities.canDelta
+  },
+  {
+    id: 'mender-cli',
+    title: 'Mender CLI',
+    getLocations: defaultLocationFormatter,
+    canAccess: () => true,
+    osList: ['darwin', 'linux']
+  },
+  {
+    id: 'mender-configure-module',
+    packageId: 'mender-configure',
```

Comment:
We shortly discuss this on our call, but just as a heads-up.

With these structures we have the limitation of one component = one package. For example `mender-configure` component publishes three packages: `mender-configure`, `...-demo`, `...-timezone`; or `mender` also produces two packages. While these extra packages that we have today have limited value for this Downloads section in the GUI, we might need others in the future.

Reply:
this should now be supported (I added `demo` + `timezone` for `mender-configure` + `dev` for `mender`)